virtcontainers/monitor.go

```diff
@@ -83,7 +83,14 @@ func (m *monitor) notify(err error) {
 	}()
 
 	for _, c := range m.watchers {
-		c <- err
+		// throw away message can not write to channel
+		// make it not stuck, the first error is useful.
+		select {
+		case c <- err:
+
+		default:
+			virtLog.Warnf("watcher channel is full, throw notify error: %v", err)
```

Comment:
Again, not required but it might be useful to log the channel size:


```go
const watcherChannelSize = 128

virtLog.WithError(err).WithField("channel-size": watcherChannelSize).Warn("watcher channel is full");
```